### PR TITLE
Prototype  sensitive terms metric

### PR DESCRIPTION
**Pull request by jsturek**
Start of prototype still need define default and custom UNL list of terms and identify or define help page for how to handle/address sensitive terms found (see `src/Metric.php ~line 85`).

For now just need to verify works how we want and verify the language used in the copy.

### CODE REVIEW

File: src/Metric.php [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+     */
+    public function scan($uri, \DOMXPath $xpath, $depth, Page $page, Metrics $context)
+    {
+        $occurrences = $this->getSensitiveTermsOccurrences($xpath);
+        $mark = $this->getMark(
+            $this->getMachineName(),
+            'A sensitive term was found.',
+            0,
+            'Verify if word is appropriate for its context, and replace if not.',
+            'See [the WDN style guide documentation](https://wdn.unl.edu/unl-web-framework-5-standards-guide) for help on this topic.',
```
Comment (jsturek) [inline src/Metric.php]: Need to identify or create help page how to handle sensitive terms.

File: src/Metric.php [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+                        }
+                    }
+                }
+                fclose($handle);
+            }
+        } catch (Exception $e) {
+            // Do Nothing
+        }
+    }
+}
```
Comment (skoolbus39) [inline src/Metric.php]: No newline at end of file

File: src/Plugin.php [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+     *
+     * @return array
+     */
+    function getEventListeners()
+    {
+        $listeners = array();
+
+        return $listeners;
+    }
+}
```
Comment (skoolbus39) [inline src/Plugin.php]: No newline at end of file

File: src/terms_list/.gitignore [hunk 4]
```diff
@@ -0,0 +1,3 @@
+*
+!default.csv
+!.gitignore
```
Comment (skoolbus39) [inline src/terms_list/.gitignore]: No newline at end of file

File: src/terms_list/default.csv [hunk 5]
```diff
@@ -0,0 +1,12 @@
...
+pimp
+got balls,has balls,takes balls
+off the reservation
+ghetto
+sack up
+insane
+goddamnit,goddammit,goddamn,goddam,oh my God
+sexy
+slave
+you guys, these guys, my guys, those guys, hey guys, hi guys, the guys, these guys, thanks guys, guys?
```
Comment (skoolbus39) [inline src/terms_list/default.csv]: No newline at end of file

File: tests/MetricTest.php [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+        $this->assertEquals(trim($expectedResult[1]['context']), trim($result[1]['context']));
+    }
+
+    public function getTestXPath($filename)
+    {
+        $parser = new HTMl5();
+        $html = file_get_contents(__DIR__ . '/data/' . $filename);
+        return $parser->parse($html);
+    }
+}
```
Comment (skoolbus39) [inline tests/MetricTest.php]: No newline at end of file

File: tests/data/sensitive_terms.html [hunk 7]
```diff
@@ -0,0 +1,11 @@
...
+<html lang='en'>
+<head>
+    <title>SiteMaster crazy testing site</title>
+</head>
+<body>
+<div>
+    This insane site just contains a few pages to help with integration testing of SiteMaster
+</div>
+</body>
+</html>
```
Comment (skoolbus39) [inline tests/data/sensitive_terms.html]: No newline at end of file

**Review by skoolbus39 — DISMISSED** [review]

**Pull request merged**